src/main/java/com/statful/sender/MetricsHolder.java

```diff
@@ -64,26 +71,40 @@ public final boolean addMetric(final DataPoint dataPoint) {
         }
 
         boolean inserted = this.buffer.offer(dataPoint);
+
+        this.shouldFlush(inserted);
+
         if (!inserted) {
             LOGGER.warn("metric could not be added to buffer, discarding it {} ", dataPoint.toMetricLine());
         }
         return inserted;
     }
 
     /**
+     * If a metric could not be added to the buffer tries to flush the buffer.
+     * Also check if a the buffer has at least as many items as the flush size and tries to flush the buffer
+     */
+    private void shouldFlush(final boolean inserted) {
```

Comment:
I'd rename this to be a "non-boolean" method name as it sounds a bit confusing, when I first read it I was expecting it not to have any side effects and to just return a boolean.

Something like `flushOnCapacity` would be a bit better I think.